### PR TITLE
fix(projects): Show join a team if person doesn't have a team

### PR DESCRIPTION
**Pull request by sentaur-athena**
This case is not happening in normal Sentry use but it came up with fly partnership where users can create projects without coming to Sentry at all. These people have access to projects that they or their team members deployed from fly but right now our projects tab loads empty because they have no team and project tab is filtered by team.

Fix: show them the placeholder to join a team

(There will be another fix where we add users to a team automatically at the time of provisioning but if they don't provision or we can't find the right team we still need to show this view)

Screenshot 2023-08-23 at 10 17 19 AM

### CODE REVIEW

File: static/app/components/noProjectMessage.tsx [hunk 1]
```diff
@@ -39,7 +40,7 @@ function NoProjectMessage({
       ? !!projects?.some(p => p.hasAccess)
       : !!projects?.some(p => p.isMember && p.hasAccess);
 
-  if (hasProjectAccess || !projectsLoaded || !teamsLoaded) {
+  if (isTeamMember && (hasProjectAccess || !projectsLoaded || !teamsLoaded)) {
     return <Fragment>{children}</Fragment>;
```
Comment (sentaur-athena) [inline static/app/components/noProjectMessage.tsx:44]: If someone doesn't have team "children" has no projects. So we should not return children, instead we should return from the next "return" which returns the "join a team" view

File: static/app/views/alerts/index.spec.tsx [hunk 2]
```diff
@@ -1,8 +1,14 @@
 import {render, screen} from 'sentry-test/reactTestingLibrary';
 
+import TeamStore from 'sentry/stores/teamStore';
 import AlertsContainer from 'sentry/views/alerts';
 
 describe('AlertsContainer', function () {
+  beforeEach(() => {
```
Comment (sentaur-athena) [inline static/app/views/alerts/index.spec.tsx:7]: The reason for the unrelated change from here is that all these components use the component I tough and when writing tests they just assumed everyone has at least one team.

**Review by cathteng — COMMENTED** [review]
are the projects being created attached to a team? they have to join the correct team to be able to see their project, right?

**Comment by sentaur-athena** [comment]
@cathteng, not necessarily. The view I'm editing (screenshot above) has two buttons, project and team. Before this change :
1. If you didn't have team or project by default it would tell you to "create a project". But you can't create project without being on a team, so I replaced the default there with "join a team"
2. If you had projects but no team (this only happens for fly users) the view showed nothing, because it only shows up if person doesn't have any project. But it also doesn't show any project because this view is filtered by teams. So in that case I show "join a team" button again.

**Comment by cathteng** [comment]
@sentaur-athena but fly users will need to attach their projects to a team to be able to see them on this page too? even after a user joins a team?

**Comment by sentaur-athena** [comment]
@cathteng their projects are attached to a team but they're not member of that team 😆 Here's the flow:

1. Org owner calls our API in Fly and creates an org
2. Fly org member (who is not on Sentry) then can add projects to that org through fly -> this person would not have problem because our API adds them to the org team
3. Fly org member can also just call Fly API to see issues. This is the problematic user. Because they had no interactions with Sentry before, have no project and no teams but should be able to see all issues according to fly logic. Our workaround is to show this person -> join a team button

**Review by cathteng — APPROVED** [review]
